src/xcode/ENA/ENA/Source/Scenes/Events/OnBehalfCheckinSubmission/DateTimeSelection/OnBehalfDateTimeSelectionViewModel.swift

```diff
@@ -0,0 +1,170 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+import Contacts
+import OpenCombine
+
+final class OnBehalfDateTimeSelectionViewModel {
+
+	// MARK: - Init
+
+	init(
+		traceLocation: TraceLocation,
+		onPrimaryButtonTap: @escaping (Checkin) -> Void
+	) {
+		self.traceLocation = traceLocation
+		self.onPrimaryButtonTap = onPrimaryButtonTap
+
+		if let startDate = traceLocation.startDate, startDate.timeIntervalSince1970 > 0 {
+			selectedDate = startDate
+		} else {
+			selectedDate = Date()
+		}
+
+		selectedDuration = TimeInterval(traceLocation.suggestedOnBehalfWarningDurationInMinutes * 60)
+	}
+
+	// MARK: - Internal
+
+	func createCheckin() {
+		let checkin: Checkin = Checkin(
+			id: 0,
+			traceLocationId: traceLocation.id,
+			traceLocationIdHash: traceLocation.idHash ?? Data(),
+			traceLocationVersion: traceLocation.version,
+			traceLocationType: traceLocation.type,
+			traceLocationDescription: traceLocation.description,
+			traceLocationAddress: traceLocation.address,
+			traceLocationStartDate: traceLocation.startDate,
+			traceLocationEndDate: traceLocation.endDate,
+			traceLocationDefaultCheckInLengthInMinutes: traceLocation.defaultCheckInLengthInMinutes,
+			cryptographicSeed: traceLocation.cryptographicSeed,
+			cnPublicKey: traceLocation.cnPublicKey,
+			checkinStartDate: selectedDate,
+			checkinEndDate: selectedDate.addingTimeInterval(selectedDuration),
+			checkinCompleted: false,
+			createJournalEntry: false,
+			checkinSubmitted: false
+		)
+
+		onPrimaryButtonTap(checkin)
+	}
+
+	var dynamicTableViewModel: DynamicTableViewModel {
+		DynamicTableViewModel([
+			.section(
+				cells: [
+					traceLocationCell(),
+					.subheadline(
+						text: AppStrings.OnBehalfCheckinSubmission.DateTimeSelection.description,
+						color: .enaColor(for: .textPrimary2)
+					),
+					dateSelectionCell(),
+					.space(height: 8),
+					durationSelectionCell()
+				]
+			)
+		])
+	}
+
+	// MARK: - Private
+
+	private let traceLocation: TraceLocation
+	private let onPrimaryButtonTap: (Checkin) -> Void
+
+	private var selectedDate: Date
+	private var selectedDuration: TimeInterval
+
+	private var durationSelectionCollapsed = true
+	private var dateSelectionCollapsed = true
+
+	private func traceLocationCell() -> DynamicCell {
+		DynamicCell.custom(
+			withIdentifier: EventTableViewCell.dynamicTableViewCellReuseIdentifier,
+			accessoryAction: .none
+		) { [weak self] _, cell, _ in
+			guard let self = self, let traceLocationCell = cell as? EventTableViewCell else {
+				return
+			}
+
+			traceLocationCell.configure(
+				cellModel: OnBehalfTraceLocationCellModel(traceLocation: self.traceLocation),
+				onButtonTap: {}
```

Comment:
Is this implementation left empty on purpose?

Reply:
Yes, in this configuration the button is hidden, that's why the closure is empty.